### ISSUE TITLE
Training code

### ISSUE THREAD

**Issue by tpetri94** [Rishit-dagli/Perceiver#22]
Hi there,

I've tried to set up a standard MNIST training over the last few days using the Perceiver code provided here. 
So far, I've not been able to come up with any solution where the model actually learns anything. A major problem so far has been the way the model is written with no support for model.fit() and the whole functional API.

Do you happen to have any training example code for your model which you could provide here in this repo?
MNIST as the default starting point would be nice, but anything would do the job as well :)

**Comment by Rishit-dagli** [comment]
Hi @tpetri94 ,
No, as of now the repo doesn't contain a training example. However, I believe it is quite simple to do so, maybe take some inspiration to do so from this [alternate implementation](https://keras.io/examples/vision/perceiver_image_classification/#compile-train-and-evaluate-the-mode) from Keras Examples. However, in this example too with almost 45 epochs on a Tesla V100, it achieves 52% accuracy on MNIST, similar to this with `model.fit()`.

**Comment by Rishit-dagli** [comment]
I am closing this issue, feel free to reach back
